### PR TITLE
Added book entity. Refactored code

### PR DESCRIPTION
**Pull request by nowicki-maciej**

### CODE REVIEW

File: core/src/main/java/cf/dmms/app/core/book/BookService.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package cf.dmms.app.core.book;
+
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class BookService {
```
Comment (michal-krajewski) [inline core/src/main/java/cf/dmms/app/core/book/BookService.java:10]: No way to update book?

File: flavors/web-app/src/main/java/cf/dmms/app/web/api/author/AuthorMapper.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class AuthorMapper {
+
+//	static AuthorDto toDto(Author author) {
+//		return AuthorDto.from(
+//				author.getId(),
+//				author.getName(),
+//				author.getSurname()
+//		);
+//	}
```
Comment (michal-krajewski) [inline flavors/web-app/src/main/java/cf/dmms/app/web/api/author/AuthorMapper.java]: ?

File: flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.List;
+
+class BookDto {
+
+	private Long id;
+	private String title;
+	private String isbn;
+	private String description;
+	private List<Long> authors;
```
Comment (michal-krajewski) [inline flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java]: Do we want to return id's? Maybe some kind of dto?

**Review by michal-krajewski — CHANGES_REQUESTED** [review]

File: flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java (hunk 3 above)
Comment (nowicki-maciej) [reply flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java]: What DTO do you propose?

Comment (nowicki-maciej) [reply flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java]: same as Author? ID, name, surname?

File: core/src/main/java/cf/dmms/app/core/book/BookService.java (hunk 1 above)
Comment (nowicki-maciej) [reply core/src/main/java/cf/dmms/app/core/book/BookService.java:10]: I decided to leave it for now until we know how we want to handle entity changes

File: flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java (hunk 3 above)
Comment (michal-krajewski) [reply flavors/web-app/src/main/java/cf/dmms/app/web/api/book/BookDto.java]: Actually Author also has list of his books. So proposed by You dto seems to be ok.

File: core/src/main/java/cf/dmms/app/core/author/AuthorService.java [hunk 4]
```diff
@@ -11,29 +13,27 @@ public AuthorService(AuthorRepository authorRepository) {
...
-		author = authorRepository.save(author);
-		authorDto.setId(author.getId());
-
-		return authorDto;
+	public Author createAuthor(Author author) {
+		return authorRepository.save(author);
 	}
 
-	public AuthorDto updateAuthor(Long authorId, AuthorDto authorDto) {
+	public Author updateAuthor(Long authorId, Author changedAuthor) {
```
Comment (michal-krajewski) [inline core/src/main/java/cf/dmms/app/core/author/AuthorService.java]: We have here Author and his id. Can't we just override existing one with this changed one?

**Review by michal-krajewski — APPROVED** [review]

**Pull request merged**